### PR TITLE
VSH: Stubs

### PR DESCRIPTION
**Pull request by elad335**
This was a bit of a PITA but it should be worth it.

### CODE REVIEW

File: rpcs3/Emu/CMakeLists.txt [hunk 1]
```diff
@@ -173,6 +173,16 @@ target_sources(rpcs3_emu PRIVATE
 	Cell/lv2/sys_tty.cpp
 	Cell/lv2/sys_usbd.cpp
 	Cell/lv2/sys_vm.cpp
+	Cell/lv2/sys_hid.cpp
```
Comment (Megamouse) [inline rpcs3/Emu/CMakeLists.txt]: i think these should be added ordered by name, at least that's how it is usually

File: rpcs3/Emu/Cell/lv2/sys_process.cpp [hunk 2]
```diff
@@ -362,3 +362,11 @@ void _sys_process_exit2(ppu_thread& ppu, s32 status, vm::ptr<sys_exit2_param> ar
 
 	ppu.state += cpu_flag::dbg_global_stop;
 }
+
+error_code sys_process_spawns_a_self2(vm::ptr<u32> pid, u32 primary_prio, u64 flags, vm::ptr<void> stack, u32 stack_size, u32 mem_id, vm::ptr<void> param_sfo, vm::ptr<void> dbg_data)
+{
+	sys_process.todo("sys_process_spawns_a_self2(pid=*0x%x, primary_prio=0x%x, flags=0x%llx, stack=*0x%x, stack_size=0x%x, mem_id=0x%x, param_sdo=*0x%x, dbg_data=*0x%x"
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_process.cpp]:
```suggestion
	sys_process.todo("sys_process_spawns_a_self2(pid=*0x%x, primary_prio=0x%x, flags=0x%llx, stack=*0x%x, stack_size=0x%x, mem_id=0x%x, param_sfo=*0x%x, dbg_data=*0x%x"
```

File: rpcs3/Emu/Cell/lv2/sys_ss.h [hunk 3]
```diff
@@ -13,3 +13,10 @@ error_code sys_ss_random_number_generator(u32 arg1, vm::ptr<void> buf, u64 size)
 error_code sys_ss_access_control_engine(u64 pkg_id, u64 a2, u64 a3);
 s32 sys_ss_get_console_id(vm::ptr<u8> buf);
 s32 sys_ss_get_open_psid(vm::ptr<CellSsOpenPSID> ptr);
+error_code sys_ss_appliance_info_manager(u32 code, vm::ptr<u8> buffer);
+error_code sys_ss_get_cache_of_product_mode(vm::ptr<u8> ptr);
+error_code sys_ss_secure_rtc(u64 cmd, u64 a2, u64 a3, u64 a4);
+error_code sys_ss_get_cache_of_flash_ext_flag(vm::ptr<u64> flag);
+error_code sys_ss_get_boot_device(vm::ptr<u64> dev);
+error_code sys_ss_update_manager(u64 pkg_id, u64 a1, u64 a2, u64 a3, u64 a4, u64 a5, u64 a6);
+error_code sys_ss_virtual_trm_manager(u64 pkg_id, u64 a1, u64 a2, u64 a3, u64 a4);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_ss.h]: newline

File: rpcs3/Emu/Cell/lv2/sys_storage.cpp [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+error_code sys_storage_async_cancel()
+{
+	sys_storage.todo("sys_storage_async_cancel()");
+
+	return CELL_OK;
+}
+
+error_code sys_storage_get_device_info(u64 device, vm::ptr<StorageDeviceInfo> buffer)
+{
+	sys_storage.todo("sys_storage_get_device_info(device=0x%x, config=0x%x)", device, buffer);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_storage.cpp]:
```suggestion
	sys_storage.todo("sys_storage_get_device_info(device=0x%x, buffer=0x%x)", device, buffer);
```

File: rpcs3/Emu/Cell/lv2/sys_storage.cpp [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+error_code sys_storage_report_devices(u32 storages, u32 start, u32 devices, vm::ptr<u64> device_ids)
+{
+	sys_storage.todo("sys_storage_report_devices(storages=0x%x, start=0x%x, devices=0x%x, device_ids=0x%x)", storages, start, devices, device_ids);
+
+	return CELL_OK;
+}
+
+error_code sys_storage_configure_medium_event(u32 fd, u32 equeue_id, u32 c)
+{
+	sys_storage.todo("sys_storage_configure_medium_event(fd=0x%x, equeue_id=0x%x, 0x%x)", fd, equeue_id, c);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_storage.cpp]:
```suggestion
	sys_storage.todo("sys_storage_configure_medium_event(fd=0x%x, equeue_id=0x%x, c=0x%x)", fd, equeue_id, c);
```

File: rpcs3/Emu/Cell/lv2/sys_hid.cpp [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+error_code sys_hid_manager_ioctl(u32 hid_handle, u32 pkg_id, vm::ptr<void> buf, u64 buf_size)
+{
+	sys_hid.todo("sys_hid_manager_ioctl(hid_handle=0x%x, pkg_id=0x%llx, buf=*0x%x, buf_size=0x%llx)", hid_handle, pkg_id, buf, buf_size);
+
+	return CELL_OK;
+}
+
+error_code sys_hid_manager_check_focus()
+{
+    sys_hid.todo("sys_hid_manager_check_focus()");
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_hid.cpp]: indentation mistake.

File: rpcs3/Emu/Cell/lv2/sys_hid.cpp [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+error_code sys_hid_manager_514(u32 pkg_id, vm::ptr<void> buf, u64 buf_size)
+{
+	sys_hid.todo("sys_hid_manager_514(pkg_id=0x%x, buf=*0x%x, buf_size=0x%llx)", pkg_id, buf, buf_size);
+
+	return CELL_OK;
+}
+
+error_code sys_hid_manager_is_process_permission_root(u32 pid)
+{
+    sys_hid.todo("sys_hid_manager_is_process_permission_root(pid=0x%x)", pid);
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_hid.cpp]: indentation mistake.

File: rpcs3/emucore.vcxproj [hunk 8]
```diff
@@ -69,6 +69,7 @@
     </PreBuildEvent>
   </ItemDefinitionGroup>
   <ItemGroup>
+    <ClCompile Include="Emu\Cell\lv2\sys_crypto_engine.cpp" />
```
Comment (elad335) [inline rpcs3/emucore.vcxproj]: manually move this.

File: rpcs3/Emu/Cell/lv2/sys_btsetting.h [hunk 9]
```diff
@@ -0,0 +1,8 @@
+﻿#pragma once
+
+#include "stdafx.h"
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_btsetting.h]: I think this is not needed

Comment (kd-11) [reply rpcs3/Emu/Cell/lv2/sys_btsetting.h]: It's also undesirable to add a pch to a header file.

Comment (Nekotekina) [reply rpcs3/Emu/Cell/lv2/sys_btsetting.h]: Probably meant to include Types.h or something, not a big problem.

**Comment by elad335** [comment]
All review comments resolved.

**Pull request merged**